hls-graph/src/Development/IDE/Graph/Internal/Action.hs

```diff
@@ -120,7 +120,7 @@ apply ks = do
     stack <- Action $ asks actionStack
     (is, vs) <- liftIO $ build db stack ks
     ref <- Action $ asks actionDeps
-    liftIO $ modifyIORef ref (ResultDeps (fromListKeySet $ toList is) <>)
+    liftIO $ modifyIORef' ref (ResultDeps [fromListKeySet $ toList is] <>)
```

Comment:
I think we need to force `fromListKeySet $ toList is` before putting it in the list.

Reply:
make sense, I added derving NFData to KeySet and force it here.

Reply:
I am not sure if we need something like 
```
    kv <- liftIO $ evaluate $ force $ fromListKeySet $ toList is
    liftIO $ modifyIORef' ref (ResultDeps [kv] <>)
```
to ensure the force?

Reply:
just `let !kv = fromListKeySet $ toList is` should be sufficient.